source/common/http/http2/codec_impl.cc

```diff
@@ -127,9 +127,13 @@ ProdNghttp2SessionFactory::create(const nghttp2_session_callbacks* callbacks,
                                   ConnectionImpl* connection, const nghttp2_option* options) {
   auto visitor = std::make_unique<http2::adapter::CallbackVisitor>(
       http2::adapter::Perspective::kClient, *callbacks, connection);
-  http2::adapter::Http2VisitorInterface& v = *visitor;
+  auto adapter = http2::adapter::NgHttp2Adapter::CreateClientAdapter(*visitor, options);
```

Comment:
does this help address the leak?  if so should it have a regression test?
/wait-any

Reply:
Done.